src/sparsematrix.jl

```diff
@@ -212,6 +212,7 @@ nnz1(S) = sum(length.(nzrange.(Ref(S), axes(S, 2))))
 function count(pred, S::AbstractSparseMatrixCSC)
     count(pred, nzvalview(S)) + pred(zero(eltype(S)))*(prod(size(S)) - nnz(S))
 end
+count(S::AbstractSparseMatrixCSC{Bool}) = count(identity, S)
```

Comment:
Doesn't these exist in Base already?

Reply:
In Base, they dispatch to `count(identity, S; dims=:, init=0)` and that goes further down the dense route:

```julia
count(A::AbstractArrayOrBroadcasted; dims=:, init=0) = count(identity, A; dims, init)
count(f, A::AbstractArrayOrBroadcasted; dims=:, init=0) = _count(f, A, dims, init)

_count(f, A::AbstractArrayOrBroadcasted, dims::Colon, init) = _simple_count(f, A, init)
_count(f, A::AbstractArrayOrBroadcasted, dims, init) = mapreduce(_bool(f), add_sum, A; dims, init)
```

So, for `dims=:`, this calls `_simple_count`. We could perhaps hook into

```suggestion
Base._count(f, A::AbstractSparseMatrixCSC, ::Colon, _) = count(f, A)
```

if that looks better?

Reply:
Okay, looks a bit scary perhaps. In any case, the arguments shouldn't be ignored I guess? I.e. if `init` is nonzero it should be added?

Reply:
Sounds correct, yes. So perhaps we need to adjust the signatures incl. the keyword arguments to the dense case then?